### PR TITLE
feat(ENG_1044): node check impl

### PR DESCRIPTION
**Pull request by MacLikorne**

### CODE REVIEW

File: src/cloud_provider/kubernetes.rs [hunk 1]
```diff
@@ -704,6 +732,38 @@ pub fn check_kubernetes_has_enough_resources_to_deploy_environment(
...
+) -> Result<(), SimpleError>
+where
+    P: AsRef<Path>,
+{
+    return match kubectl_exec_get_node(kubernetes_config, envs) {
+        Err(e) => {
+            error!("Can't get nodes before services deployments");
+            Err(e)
+        }
+        Ok(nodes) => match nodes.items.len() < max_nodes * 80 / 100 {
```
Comment (deimosfr) [inline src/cloud_provider/kubernetes.rs:749]: please add () to avoid any bad missunderstanding

File: src/cloud_provider/kubernetes.rs [hunk 2]
```diff
@@ -704,6 +732,38 @@ pub fn check_kubernetes_has_enough_resources_to_deploy_environment(
...
+) -> Result<(), SimpleError>
+where
+    P: AsRef<Path>,
+{
+    return match kubectl_exec_get_node(kubernetes_config, envs) {
+        Err(e) => {
+            error!("Can't get nodes before services deployments");
+            Err(e)
+        }
+        Ok(nodes) => match nodes.items.len() < (max_nodes * 80 / 100) {
```
Comment (erebe) [inline src/cloud_provider/kubernetes.rs:749]: doesn't it break autoscaler ?
if you have a cluster with [3 min, 10 max] nodes,
and say you are at 3 nodes ATM, you will never be able to scale your cluster no ?

Comment (erebe) [inline src/cloud_provider/kubernetes.rs:749]: if you are already at max node, let's say 3;3. you will never be able to deploy something no ?

Comment (MacLikorne) [reply src/cloud_provider/kubernetes.rs:749]: yes that's the point, if more than 80% of nodes capacity is used, it's stuck

Comment (erebe) [reply src/cloud_provider/kubernetes.rs:749]: ok for infra, but for env you will never be able to deploy something has you will have 3 nodes, but they will be empty

Comment (deimosfr) [reply src/cloud_provider/kubernetes.rs:749]: If you're 3/3 yes, but you can be sure that at least 50% of the time you won't be able to deploy. So, the user will just have to upgrade the max capacity. That won't mean the cluster capacity will grow, but we reduce issues due to this point.

**Pull request closed without merging**